### DISCUSSION TITLE
Windows Subsystem for Linux WSLg Preview - 1.0.27 Won't Install under Windows 11 22H2

### DISCUSSION THREAD

**Discussion by ourichermath** [microsoft/wslg#876] [Q&A] [answered]
Windows Update won't install Windows Subsystem for Linux WSLg Preview - 1.0.27.  It appears to download and install, but then just re-appears with "Install error - 0x80070643".  The Event log gets two errors for the attempt:

Event 20, WindowsUpdateClient:
"Installation Failure: Windows failed to install the following update with error 0x8024200B: Windows Subsystem for Linux WSLg Preview - 1.0.27."

and

Event 11714, MsiInstaller:
"Product: Windows Subsystem for Linux WSLg Preview -- Error 1714. The older version of Windows Subsystem for Linux WSLg Preview cannot be removed.  Contact your technical support group.  System Error 1612."

Surprisingly, since I don't think I ever installed WSLg, version 1.0.26 is showing as installed in my App list.  I think what happened is that I'd installed the WSL Preview a while back, and it somehow updated to include WSLg 1.0.26 (and why it's trying to download the WSLg Preview).  Regardless, when I try to uninstall it, it fails with the following:

"The path 'C:\Users\[username]\AppData\Local\Temp\wsl_graphics_support_x64.msi' cannot be found.  Verify that you have access to this location and try again, or try to find the installation package "wsl_graphics_support_x64.msi' in a folder from which you can install the product Windows Subsystem for Linux WSLg Preview."

So, I'd guess the Windows Update update to WSLg won't work because it can't find the uninstall routine to delete the previous version.

I also uninstalled the plain old WSL Preview App I had, installed it again in Terminal with:

`wsl --install -d Ubuntu`

updated its kernel with:

`wsl update`

Then I did a "Clean Up System Files" in Disk Cleanup and cleared Windows Upgrade Log Files and Deliver Optimization Files.  After various reboots and Safe Boots, I tried running Windows Update again and it gave me the option to install the Windows Subsystem for Linux Update (5.10.102.1) and that worked.  But, the WSLg Preview still won't install or go away.  I've uninstalled the plain WSL Preview (again) and am just left with the plain WSL Update and that old WSLg 1.0.26 still can't install or get rid of the 1.0.27 Preview.

Is there some other way to get rid of the old 1.0.26 WSLg version even though I don't have its installation program?  I don't even care if I get the Preview.  I really just want to clear it out of Windows Update and wait for the released version.

**Comment by ourichermath** [comment, marked as answer]
I finally fixed this by downloading the 1.0.26 wsl_graphics_support_x64.msi from here, uninstalling the app by pointing the uninstaller to that file, and then installing the 1.0.27 version stuck in Windows Update.

**Reply by ansemjo** [reply]
Thank you!

Another note: the file you select must be called exactly `wsl_graphics_support_x64.msi`. I wasn't sure which MSI installer to use and picked the wrong one at first, so that the 1.0.26 installer file was called `wsl_graphics_support_x64(1).msi` ... the uninstaller rejected that file until I renamed it. 🙃

**Reply by YoadHamama** [reply]
What do you mean by the uninstaller? You mean via control panel?

**Reply by killTheHostage** [reply]
I found another solution: on my computer, WSLg installation failed because the key "\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Terminal Server Client\Default\OptionalAddIns\WSLDVC" failed to be written. The original key on my computer is "\HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Terminal Server Client\Default\OptionalAddIns\WSLDVC_PACKAGE", remove the "_PACKAGE" and you can install it smoothly.

**Reply by RAASORG** [reply]
esto me funcionó

**Reply by EdenSeven** [reply]
it works for me

**Reply by LucasCordova** [reply]
This worked for me!

**Reply by killTheHostage** [reply]
Update: If you use this method, you will need to add back the removed "_PACKAGE" value when you upgrade to wsl2.0
